Ignore MacOS' auxilary directories

hubertp opened the pull request:
### Pull Request Description

It appears that on some MacOS CI runners some indexing is turned on, which means that OS creates automatically hidden directories. This messes up with our expectations. This change adds an exception for MacOS to simply ignore `._` prefixed directories during unpacking in `Archive.scala`.

The rest of the changes is just to move `Platform` logic to `os-environment` package where it should be in the first place.

Closes #14294.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.

hubertp committed:
Ignore MacOS' auxilary directories

It appears that on some MacOS CI runners some indexing is turned on,
which means that OS creates automatically hidden directories. This
messes up with our expectations. This change adds an exception for MacOS
to simply ignore `._` prefixed directories during unpacking in
`Archive.scala`.

The rest of the changes is just to move `Platform` logic to
`os-environment` package where it should be in the first place.

hubertp committed:
typo

hubertp committed:
fix std-benchmarks

JaroslavTulach commented on lib/java/os-environment/src/main/java/org/enso/os/environment/Platform.java:
- Why did #12618 decided to move the class to common?
- Done by this commit https://github.com/enso-org/enso/pull/12618/commits/a923150a00b4d88554d864c21ef5310a50f09482
- any clue, @Akirathan?

Akirathan replied:
Looking at my private notes from that time:
- `Platform` should be migrated from `desktop-environment` to `engine-common`
- `aproject-manager` already depends transitivelly on `engine-common`
- `engine-runner` depends on `engine-common`
- Add new dependency `desktop-environment` --> `engine-common`
- `Platform` will be shared between `desktop-environment` and `engine-runner`.

Since both `desktop-environment` and `project-manager` were effectively removed, my reasons now seem outdated. So if @hubertp manages to get all tests green, I am not opposed this refactoring.

JaroslavTulach replied:
> Since both `desktop-environment` and `project-manager` were effectively removed, 
> my reasons now seem outdated

I'll try to add one more argument in favor of not doing this refactoring:
- `os-environment` is _native image_ module full of _system Java_ calls like 
- https://github.com/enso-org/enso/blob/900bc3b5dc584ffecc9a087203176b2c48a90faf/lib/java/os-environment/src/main/java/org/enso/os/environment/chdir/PosixWorkingDirectory.java#L77
- the module is only usable in _native image_ mode
- that's not the case of `Platform` - it is fully usable in JVM mode:
- https://github.com/enso-org/enso/blob/900bc3b5dc584ffecc9a087203176b2c48a90faf/engine/common/src/main/java/org/enso/common/Platform.java#L3

maybe that was another reason why `Platform` got separated and moved to `engine-common` and why I feel it should stay there.

hubertp replied:
And make `downloader` depend on `engine-common`? This really just feels wrong.

While this change has opened a can of worms, it has also revealed some interesting things. Like `runtime-language-epb` not being very minimal at all and including all `runtime` as a dependency. See https://github.com/enso-org/enso/pull/14340/commits/e0195f61ced3fd3bfbed6ee63779f3b9cc253c42

JaroslavTulach replied:
> And make `downloader` depend on `engine-common`? This really just feels wrong.

What's wrong on that?  Does it feel wrong because it is called "engine"? Would renaming the project solve the "wrongness"?

> While this change has opened a can of worms

Yes, it did. I'd say this PR has _lost its direction_. It is no longer about **Ignore MacOS' auxiliary directories**. It is about refactoring dependencies and I am not sure I like where it is heading.

> it has also revealed some interesting things. Like runtime-language-epb not being very minimal

- it is certainly my goal to have dependencies under control
   - I added various [*DependenciesTest](https://github.com/enso-org/enso/blob/7898863fb49857f8084a60931d9d83f5cc110866/engine/runtime-language-epb/src/test/java/org/enso/interpreter/epb/EpbLanguageDependenciesTest.java#L5) around the code base
- if there is a better way to control dependencies in `sbt`, then I am all for more tight control
- however I don't understand e0195f6 - is `sbt dependencyInfo` giving accurate results!?
- the way to see real (e.g. `javac`) dependencies of EPB language is:
```
$ cat engine/runtime-language-epb/.enso-sources-classes

options.0=--module-path
options.1=/maven2/org/graalvm/truffle/truffle-api/25.0.1/truffle-api-25.0.1.jar\:
/maven2/org/graalvm/sdk/collections/25.0.1/collections-25.0.1.jar\:
/maven2/org/graalvm/sdk/nativeimage/25.0.1/nativeimage-25.0.1.jar\:
/maven2/org/graalvm/polyglot/polyglot/25.0.1/polyglot-25.0.1.jar\:
/maven2/org/graalvm/sdk/word/25.0.1/word-25.0.1.jar\:
/enso/lib/java/jvm-channel/target/classes\:
/enso/lib/java/jvm-interop/target/classes\:
/enso/lib/java/ydoc-polyfill/target/ydoc-polyfill.jar\:
/enso/lib/java/runtime-utils/target/scala-2.13/classes

options.18=-classpath
options.19=/enso/engine/runtime-language-epb/target/scala-2.13/classes\:
/maven2/org/scala-lang/scala-library/2.13.15/scala-library-2.13.15.jar\:
/maven2/org/frgaal/compiler/24.0.0/compiler-24.0.0.jar\:
/maven2/org/graalvm/truffle/truffle-api/25.0.1/truffle-api-25.0.1.jar\:
/maven2/org/graalvm/truffle/truffle-dsl-processor/25.0.1/truffle-dsl-processor-25.0.1.jar\:
/maven2/org/graalvm/sdk/collections/25.0.1/collections-25.0.1.jar\:
/maven2/org/graalvm/sdk/nativeimage/25.0.1/nativeimage-25.0.1.jar\:
/maven2/org/graalvm/polyglot/polyglot/25.0.1/polyglot-25.0.1.jar\:
/maven2/org/graalvm/sdk/word/25.0.1/word-25.0.1.jar
```
- and such dependencies looks more or less close to what I'd like to see there!
- maybe `ydoc-polyfill` could now be removed...?

JaroslavTulach reviewed (commented):
> The rest of the changes is just to move Platform logic to os-environment package where it should be in the first place.

- Apparently @Akirathan had exactly opposite opinion in #12618 ...

hubertp committed:
fix modules

hubertp committed:
another module fun

hubertp committed:
Cleanup more module problems and fix epb

Previously a test in `runtime-language-epb` would blow up because it's
dependencyTree has some unexpected entries
```
sbt:enso> runtime-language-epb/dependencyTree
[info] org.enso:cli_2.13:0.1
[info] org.enso:common-polyglot-core-utils_2.13:0.1.0-SNAPSHOT
[info] org.enso:connected-lock-manager-server_2.13:0.1.0-SNAPSHOT
[info] org.enso:connected-lock-manager_2.13:0.1.0-SNAPSHOT
[info] org.enso:distribution-manager_2.13:0.1.0-SNAPSHOT
[info] org.enso:downloader_2.13:0.1
[info] org.enso:edition-updater_2.13:0.1.0-SNAPSHOT
[info] org.enso:editions_2.13:0.1.0-SNAPSHOT
[info] org.enso:engine-common:0.2-SNAPSHOT
[info] org.enso:http-test-helper_2.13:0.1.0-SNAPSHOT
[info] org.enso:interpreter-dsl_2.13:0.1
[info] org.enso:library-manager_2.13:0.1.0-SNAPSHOT
[info] org.enso:logging-config_2.13:0.1
[info] org.enso:logging-service-logback_2.13:0.1
[info] org.enso:logging-service_2.13:0.1
[info] org.enso:logging-truffle-connector_2.13:0.1
[info] org.enso:logging-utils_2.13:0.1
[info] org.enso:os-environment-lib_2.13:0.1.0-SNAPSHOT
[info] org.enso:os-environment_2.13:0.1.0-SNAPSHOT
[info] org.enso:persistance-dsl:0.2-SNAPSHOT
[info] org.enso:pkg_2.13:0.1
[info] org.enso:polyglot-api-macros_2.13:0.1.0-SNAPSHOT
[info] org.enso:polyglot-api_2.13:0.1.0-SNAPSHOT
[info] org.enso:process-utils_2.13:0.1.0-SNAPSHOT
[info] org.enso:python-resource-provider_2.13:0.1.0-SNAPSHOT
[info] org.enso:runtime-compiler-dump-igv_2.13:0.1.0-SNAPSHOT
[info] org.enso:runtime-compiler-dump_2.13:0.1.0-SNAPSHOT
[info] org.enso:runtime-compiler_2.13:0.1.0-SNAPSHOT
[info] org.enso:runtime-language-epb_2.13:0.0.0-dev [S]
[info]   +-org.enso:jvm-interop:0.2-SNAPSHOT
[info]     +-org.enso:jvm-channel:0.2-SNAPSHOT
[info]       +-org.enso:persistance:0.2-SNAPSHOT
[info]
[info] org.enso:runtime-parser-dsl_2.13:0.2-SNAPSHOT
[info] org.enso:runtime-parser-processor_2.13:0.2-SNAPSHOT
[info] org.enso:runtime-parser:0.2-SNAPSHOT
[info] org.enso:runtime-suggestions_2.13:0.1.0-SNAPSHOT
[info] org.enso:runtime-utils_2.13:0.1.0-SNAPSHOT
[info] org.enso:runtime_2.13:0.0.0-dev
[info] org.enso:scala-libs-wrapper_2.13:0.1.0-SNAPSHOT
[info] org.enso:scala-yaml_2.13:0.1.0-SNAPSHOT
[info] org.enso:semver_2.13:0.1.0-SNAPSHOT
[info] org.enso:syntax-rust-definition:0.2-SNAPSHOT
[info] org.enso:test-utils_2.13:0.1.0-SNAPSHOT
[info] org.enso:testkit_2.13:0.1.0-SNAPSHOT
[info] org.enso:text-buffer_2.13:0.1.0-SNAPSHOT
[info] org.enso:version-output_2.13:0.1
```
This is because of ... `test-utils` dependency in `jvm-interop`.
Now that `downloader` has a dependency on `os-environment`, a test
checking for slf4j or apache commons would blow up. This change cleans
up that dependency hell a bit.

hubertp committed:
fmt

hubertp commented:
Looks like I really opened a can of worms.

Now vscode plugin will not build because the dependency hell will trigger `polyglot-api` compilation which in turn requires `flatc` being setup. 
There is plenty of problems here, like why do we even have `org.enso.os.environment.jni` in `os-environemnt` or why `runtime-language-epb` manages to have all `runtime` in its dependency tree (hint see `sbt> runtime-language-epb/dependencyTree` ). I will try to address this in a separate PR without unsettling hornets' nest. See #14369.

JaroslavTulach commented:
> trigger polyglot-api compilation

- Yes, `polyglot-api` is especially dangerous, bloated dependency.
   - see `cat engine/polyglot-api/.enso-sources-classes`
- it has to be avoided for projects like `runtime-parser` to remain portable

> manages to have all runtime in its dependency tree

- as [noted here](https://github.com/enso-org/enso/pull/14340#discussion_r2565500495) I doubt `sbt dependencyInfo` is working correctly
- yes, having a reliable **way to control dependencies** is a prerequisite to improve anything for real
